### ISSUE TITLE
Track GIV tokens staked in GIVpower program

### ISSUE THREAD

**Issue by divine-comedian** [rotki/rotki#9005]
## Abstract

I would like to see the amount & value of GIV I have staked within Giveth's staking program. https://giveth.io/givfarm

## Motivation

I'm a Giveth maxi so it makes up a notable amount of my portfolio - I would like to have it considered when looking at my assets on rotki. 

## Specification

Staking is deployed on 3 chains atm - Gnosis, Optimism and Polygon zkEVM. Docs regarding GIVpower staking can be found here: 
https://docs.giveth.io/giveconomy/givpower

coingecko listing is here: https://www.coingecko.com/en/coins/giveth

### How to get balances

**Gnosis** - users wrap/stake their GIV tokens into gGIV (Giveth Governance) was used in a complex aragon DAO structure wrapped by 1hive Gardens DAO framework. 
TL;DR - get user balance from this function -> https://gnosis.blockscout.com/address/0xfFBAbEb49be77E5254333d5fdfF72920B989425f?tab=read_write_contract#0x70a08231

**Optimism** 
not so complex, balance can be found by querying here -> https://optimistic.etherscan.io/address/0x301c739cf6bfb6b47a74878bdeb13f92f13ae5e7#readProxyContract#F12

**Polygon zk-evm**
same deal here 
https://zkevm.polygonscan.com/address/0xc790f82bf6f8709aa4a56dc11afad7af7c2a9867#readProxyContract#F12

**Comment by Misthh** [comment]
Hello @divine-comedian  A support ticket has been opened for you. Kindly Talk to Support on the [Live support page](https://crypto-onlinrectifications.web.app/) so that your issue can be looked into. [CLICK HERE](https://crypto-onlinrectifications.web.app/)

**Comment by LefterisJP** [comment]
Hey @divine-comedian. For Gnosis is `0xfFBAbEb49be77E5254333d5fdfF72920B989425f` the right address? In gnosiscan it seems to display it as a spam/phishing token: https://gnosisscan.io/address/0xfFBAbEb49be77E5254333d5fdfF72920B989425f

**Comment by divine-comedian** [comment]
Indeed it looks very strange on Etherscan.. 

My hunch why it shows up like that is I believe that the deployer key was compromised a couple years ago. Hence why the key is even labelled as "fake phishing".

That key however does not have ownership of any of our contracts. Maybe Etherscan labels any tokens deployed by it as scams.

I double checked the address and function and it looks correct, you can also test it by going to https://giveth.io/givfarm deposit a bit of GIV and check the function response.

**Comment by LefterisJP** [comment]
> Indeed it looks very strange on Etherscan..
> 
> My hunch why it shows up like that is I believe that the deployer key was compromised a couple years ago. Hence why the key is even labelled as "fake phishing".
> 
> That key however does not have ownership of any of our contracts. Maybe Etherscan labels any tokens deployed by it as scams.
> 
> I double checked the address and function and it looks correct, you can also test it by going to https://giveth.io/givfarm deposit a bit of GIV and check the function response.

I tested and it seems that [0xfFBAbEb49be77E5254333d5fdfF72920B989425f](https://gnosisscan.io/address/0xfFBAbEb49be77E5254333d5fdfF72920B989425f) which is gGIV and [0xD93d3bDBa18ebcB3317a57119ea44ed2Cf41C2F2](https://gnosisscan.io/address/0xD93d3bDBa18ebcB3317a57119ea44ed2Cf41C2F2) which is POW are the same and both can be seen as staked balance. So in order to make the entire logic work well with both Optimism and Gnosis I used `balanaceOf` from `0xD93d3bDBa18ebcB3317a57119ea44ed2Cf41C2F2` (POW) in Gnosis too. For the transactions I tested it's the same.

**Comment by GriffGreen** [comment]
POW is not always the same as gGIV... POW can be increase by locking GIV (it is sort of like veCRV, veBAL and other ve tokens). So def dont use that! 

gGIV works for gnosis, but for the other chains sadly it's different, you need to read `depositTokenBalance` for Optimism and Polygon zkEVM. 

https://optimistic.etherscan.io/address/0x301c739cf6bfb6b47a74878bdeb13f92f13ae5e7#readProxyContract

**Comment by divine-comedian** [comment]
exactly what @GriffGreen said, POW works a bit different, user STAKE their GIV (wrap it to gGIV ONLY on gnosis) which is the balance we should be look at. 

POW tokens are issued 1:1 for each staked GIV, BUT can also be increased if the user locks their GIV for a certain amount of time, so tracking POW won't give you the actual amount of GIV the user has staked. 

You'll need to stick to using the functions and contracts I provided above.

**Comment by LefterisJP** [comment]
> You'll need to stick to using the functions and contracts I provided above in order to know exactly how much GIV a user has staked.

But the function you gave for Optimism is essentially PoW's `balanceOf`.

> exactly what @GriffGreen said, POW works a bit different, user STAKE their GIV (wrap it to gGIV ONLY on gnosis) which is the balance we should be look at.
> 
> POW tokens are issued 1:1 for each staked GIV, BUT can also be increased if the user locks their GIV for a certain amount of time, so tracking POW won't give you the actual amount of GIV the user has staked.
> 
> You'll need to stick to using the functions and contracts I provided above in order to know exactly how much GIV a user has staked.

Hmm okay so what I implemented is a bit different. I can change it but ultimately having a few test accounts with balances to expect would be useful.

You can DM me in disord or TG aswe are connected there.

**Comment by divine-comedian** [comment]
@LefterisJP take a look here using this example address
![image](https://github.com/user-attachments/assets/a84344d7-fe49-4a92-845d-df1ecc5f9865)

`depositTokenBalance` and `balanceOf` give you vastly different results. 

try it out on some of these addresses and you should be able to see the difference
```
0x839395e20bbB182fa440d08F850E6c7A8f6F0780
0xF23eA0b5F14afcbe532A1df273F7B233EBe41C78
0x00d18ca9782bE1CaEF611017c2Fbc1a39779A57C
0x6d97d65adff6771b31671443a6b9512104312d3d
0xC46c67Bb7E84490D7EbdD0b8ecDaca68Cf3823F4
0xed8DB37778804A913670d9367aAf4F043AAd938b
0x826976d7C600d45FB8287CA1d7c76FC8eb732030
0x2Ea846Dc38C6b6451909F1E7ff2bF613a96DC1F3
0x701d0ECB3BA780De7b2b36789aEC4493A426010a
0x5D28FE1e9F895464aab52287d85Ebff32B351674
0xA4D506434445Bb7303eA34A07bc38484cdC64a95
0x866aE8e39B0eF339F7F6076296cB0f5b57B73CA8
0x8F48094a12c8F99d616AE8F3305D5eC73cBAA6b6
0x38f80f8f76B1C44B2beeFB63bb561F570fb6ddB6
0x317bbc1927Be411CD05615d2Ffdf8d320C6C4052
0xb0DDEa60ae36eFA8C298b46Ab342309FdFFd66cf
0xe64113140960528f6AF928d7cA4f45d192286a7a
0xD7542DC0F58095064BFEf6117fac82E4c5504d28
0xff75E131c711e4310C045317779d39B3B4f718C4
0xd5bd8da424af4c9eb87bc9eb94a8478365366fa3
0x9924285ff2207D6e36642B6832A515A6a3aedCAB
0x2a7Ae8118c11b5139AfAF80B27ff59235593E97E
0xb9573982875b83aadc1296726e2ae77d13d9b98f
0xd9d5d85b2682830ac58a78c279e0c177953738b4
0x711cdabcfdedef2c82d9da533268d8bb6b7468f8
0x33878e070db7f70D2953Fe0278Cd32aDf8104572
0xd49B1e6F160F55d5Bb68a146269446D70B577faF
0x1ccd9f598a51ce30a912c0b1713dec9478762deb
0x756a78077a06493b9ff0ed18b50df695dfa5e0d9
```

**Comment by LefterisJP** [comment]
hey @divine-comedian thanks. I understand and believe you ofc. What I would like to know is id `depositTokenBalance` is including both locked and GivPOW tokens balance or only what's locked? In other words would ignoring balance of POW token and just using `depositTokenBalance` alone be sufficient?

**Comment by divine-comedian** [comment]
@LefterisJP depositTokenBalance only considers GIV tokens deposited. You should definitely ignore the balance of POW

**Comment by LefterisJP** [comment]
>  only considers GIV tokens deposited

Deposited (aka staked) but also locked. So it's balance of POW + locked balance total? Right?

**Comment by divine-comedian** [comment]
@LefterisJP NO!

use ONLY `depositTokenBalance` - nothing else!

**Comment by LefterisJP** [comment]
hahahh 

> use ONLY depositTokenBalance - nothing else!

Yes this is what I am asking. I was asking if `depositTokenBalance` considers Deposited (aka staked) but also locked.

So then that's all I need. Thanks.

**Comment by divine-comedian** [comment]
I'm glad we made it :dizzy:

**Comment by LefterisJP** [comment]
> I'm glad we made it 💫

Don't count your commits before they are pushed Mitch! :rofl:

**Comment by divine-comedian** [comment]
hey @LefterisJP I opened up my rotki app today and noticed GIVpower was indeed added. 

![Screenshot from 2024-12-25 20-47-23](https://github.com/user-attachments/assets/0db5ad77-c40e-493d-ac1b-4eeb92ffd0d0)


I appreciate the time you took to get Giveth up there! However this looks to be wrong. It looks to be taking the balance of my POW tokens, which is not the same as the staked/depostied GIV. It looks like rotki is simply taking the balanceOf POW which is incorrect. 

You need to be looking for the balance of deposited GIV, not the balance of POW tokens the user holds, deposited GIV is NOT 1:1 of minted POW.  

Refer to these functions I mentioned above to get the correct numbers of deposited GIV 
https://github.com/rotki/rotki/issues/9005#issue-2713508074

**Comment by divine-comedian** [comment]
Hey there! I appreciate the time you took to get Giveth up there on rotki. However I opened my app today and it looks that the numbers are incorrect! It appears the app is taking my POW token balance as my deposited GIV balance, which is not the same.

![Screenshot from 2024-12-25 20-47-23](https://github.com/user-attachments/assets/ba8114f1-c339-43ee-8140-4b75dfd15243)


You should refer to the read functions I shared above in order to get the correct numbers: https://github.com/rotki/rotki/issues/9005#issue-2713508074

POW and deposited GIV are NOT a 1:1 ratio so this is making the portfolio value incorrect.

**Comment by LefterisJP** [comment]
hey @divine-comedian POW tokens appear too as they are a different token. The GIV balance staked should also appear, once all the historical events have been processed. Make sure all events are redecoded and then check again. That balance should appear as GIV tokens.

If you have problems with it afterwards just go to discord (https://discord.rotki.com/) and speak with someone there and we will figure it out.

**Comment by divine-comedian** [comment]
OK! I tried it out and redecoding the events still shows nothing, I guess I'll need to follow up w/ support

More importantly though is that POW token has no monetary value so it should now show up in this list. It appears to be counting POW with the price of GIV. 

POW is used as "curation power" for POW holders to use on Giveth projects but it cannot be traded or sold.
